convert_nndc71.py

```diff
@@ -9,126 +9,219 @@
 import argparse
 import glob
 import hashlib
-import os
 import shutil
 import subprocess
 import sys
 import tarfile
+import zipfile
+from pathlib import Path
 
 import openmc.data
 from openmc._utils import download
 
+# Make sure Python version is sufficient
+assert sys.version_info >= (3, 6), "Python 3.6+ is required"
+
+description = """
+Download ENDF/B-VII.1 incident neutron ACE data and incident photon ENDF data
+from NNDC and convert it to an HDF5 library for use with OpenMC. This data is
+used for OpenMC's regression test suite.
+"""
 
 class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
                       argparse.RawDescriptionHelpFormatter):
     pass
 
 parser = argparse.ArgumentParser(
-    description=__doc__,
+    description=description,
     formatter_class=CustomFormatter
 )
-parser.add_argument('-b', '--batch', action='store_true',
-                    help='supresses standard in')
-parser.add_argument('-n', '--neutron-only', action='store_true',
-                    help='Whether to exclude photon interaction/atomic data')
+
+parser.add_argument('-d', '--destination', type=Path, default='nndc-b7.1-hdf5',
+                    help='Directory to create new library in')
+parser.add_argument('--download', action='store_true',
+                    help='Download tarball from NNDC-BNL')
+parser.add_argument('--no-download', dest='download', action='store_false',
+                    help='Do not download tarball from NNDC-BNL')
+parser.add_argument('--extract', action='store_true',
+                    help='Extract compressed files')
+parser.add_argument('--no-extract', dest='extract', action='store_false',
+                    help='Do not extract compressed file if it has already been extracted')
 parser.add_argument('--libver', choices=['earliest', 'latest'],
                     default='earliest', help="Output HDF5 versioning. Use "
                     "'earliest' for backwards compatibility or 'latest' for "
                     "performance")
+parser.add_argument('-r', '--release', choices=['b7.1'],
```

Comment:
The NNDC is not really in the business of distributing ACE files anymore. They had done this for ENDF/B-VII.1 because there were no openly available ACE files for it otherwise. However, LANL is now releasing their libraries [openly](https://nucleardata.lanl.gov/), and they do a lot more validation and checking of their libraries than NNDC did. So, I don't foresee a need to support multiple releases for this script.

Reply:
I was half thinking about adding the latest version, release b8.0, it might be straightforward https://www.nndc.bnl.gov/endf/b8.0/download.html

Reply:
now removed release version